[Bug] TPartMin 使用语句：pto.ui32(4294967295); 运行ST报错：'arith.constant' op integer return type must be signless

### Component

PTO Dialect / ODS (include/PTO/IR)

### Description

<img width="784" height="868" alt="Image" src="https://github.com/user-attachments/assets/26a84146-9b77-4cf6-b45a-a4da3f6e8beb" />
如同所示：将26行注释打开后(注释掉运行没问题)，运行st报错如下：

<img width="892" height="393" alt="Image" src="https://github.com/user-attachments/assets/33db41eb-eae8-4dbe-9b6d-8ab23968d208" />

### Reproduction (minimal)

```markdown
st运行命令：

python test/tilelang_st/script/run_st.py -r sim -v a5 -t tpartmin -c f32_64x64_ful

template文件内容：

"""TileLang DSL template for pto.tpartmin"""

import tilelang_dsl as pto


@pto.vkernel(
    target="a5",
    op="pto.tpartmin",
    advanced=True,
)
def template_tpartmin(src0: pto.Tile, src1: pto.Tile, dst: pto.Tile):
    dtype = dst.element_type
    valid_rows, valid_cols = dst.valid_shape
    src0_valid_rows, src0_valid_cols = src0.valid_shape
    src1_valid_rows, src1_valid_cols = src1.valid_shape
    lanes = pto.get_lanes(dtype)

    pad_scalar = pto.f32(3.402823466e+38)
    if pto.constexpr(dtype == pto.f16):
        pad_scalar = pto.f16(65504.0)
    elif pto.constexpr(dtype == pto.bf16):
        pad_scalar = pto.bf16(3.3856e+38)
    elif pto.constexpr(dtype == pto.i32):
        pad_scalar = pto.i32(2147483647)
    elif pto.constexpr(dtype == pto.ui32):
        pad_scalar = pto.ui32(4294967295)
        pass
    elif pto.constexpr(dtype == pto.i16):
        pad_scalar = pto.i16(32767)
    elif pto.constexpr(dtype == pto.ui16):
        # pad_scalar = pto.ui16(65535)
        pass
    elif pto.constexpr(dtype == pto.i8):
        pad_scalar = pto.i8(127)
    elif pto.constexpr(dtype == pto.ui8):
        # pad_scalar = pto.ui8(255)
        pass

    pad_vec = pto.vbr(pad_scalar)

    for row in range(0, valid_rows, 1):
        remained = valid_cols
        for col in range(0, valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            # pto.vsts(pad_vec, dst[row, col:], mask)
            pass

    for row in range(0, src0_valid_rows, 1):
        remained = src0_valid_cols
        for col in range(0, src0_valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            vec0 = pto.vlds(src0[row, col:])
            pto.vsts(vec0, dst[row, col:], mask)

    for row in range(0, src1_valid_rows, 1):
        remained = src1_valid_cols
        for col in range(0, src1_valid_cols, lanes):
            mask, remained = pto.make_mask(dtype, remained)
            vec_dst = pto.vlds(dst[row, col:])
            vec1 = pto.vlds(src1[row, col:])
            result = pto.vmin(vec_dst, vec1, mask)
            pto.vsts(result, dst[row, col:], mask)

    return
```

### Expected behavior

编译通过

### Actual behavior / error logs

```text
expand_helper: warning: failed to import
  /home/caojian/workspace/git/github/mouliangyu/PTOAS/lib/TileOps/tmrgsort_template.py:
  parameter 'exhausted' must not declare a default value in TileLang DSL v1

  expand_helper: warning: failed to import
  /home/caojian/workspace/git/github/mouliangyu/PTOAS/lib/TileOps/tmrgsort_template.py:
  parameter 'exhausted' must not declare a default value in TileLang DSL v1

  expand_helper: warning: failed to import
  /home/caojian/workspace/git/github/mouliangyu/PTOAS/lib/TileOps/tmrgsort_template.py:
  parameter 'exhausted' must not declare a default value in TileLang DSL v1

  loc("-":25:21): error: 'arith.constant' op integer return type must be
  signless

  ExpandTileOp: failed to parse DSL output


  loc("/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/testcase/tpartmin/tpartmin.pto":591:5):
  error: ExpandTileOp: failed to instantiate tilelang template for tpartmin

  Error: backend lowering pass execution failed.


make[2]: *** [testcase/tpartmin/CMakeFiles/tpartmin_kernel.dir/build.make:85: testcase/tpartmin/tpartmin_kernel.ll] Error 1
make[2]: Leaving directory '/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/build'
make[1]: *** [CMakeFiles/Makefile2:114: testcase/tpartmin/CMakeFiles/tpartmin_kernel.dir/all] Error 2
make[1]: Leaving directory '/home/caojian/workspace/git/github/mouliangyu/PTOAS/test/tilelang_st/npu/a5/src/st/build'
make: *** [Makefile:91: all] Error 2
```

### Git commit

00535ba874929953fca21c7060bc9f975c16fb3d

### Host platform

None

### Target Ascend arch (if relevant)

None

### PTOAS build level (if relevant)

None

--- a/tilelang-dsl/docs/user_guide/05-type-system.md
+++ b/tilelang-dsl/docs/user_guide/05-type-system.md
@@ -424,6 +424,26 @@ rank = tile.rank                      # 2
 - `tile.pad_value.eval()` with `PadValue.custom_f32(...)` becomes the authored floating scalar
 - `tile.pad_value.eval()` with `PadValue.NULL` raises a frontend error
 
+For dtype-dependent fill seeds, prefer `tile.pad_value.eval()` over handwritten
+`if dtype == ...` ladders.
+
+```python
+@pto.vkernel(op="fill_pad_value", dtypes=[(pto.AnyType,)])
+def fill_pad_value(dst: pto.Tile):
+    pad_scalar = dst.pad_value.eval()
+    pad_vec = pto.vbr(pad_scalar)
+    # ...
+```
+
+Typical materialized values:
+
+- `PadValue.ZERO` -> `0` / `0.0`
+- `PadValue.MAX` -> dtype-aware max, for example `4294967295` for `pto.ui32`
+- `PadValue.MIN` -> dtype-aware min, for example `-2147483648` for `pto.i32` and `0` for `pto.ui32`
+
+This is usually simpler than spelling every dtype case manually with
+`pto.constexpr(dst.element_type == ...)`.
+
 Example: reading pad value from a `Tile`
 
 ```python

--- a/tilelang-dsl/python/tilelang_dsl/lowering.py
+++ b/tilelang-dsl/python/tilelang_dsl/lowering.py
@@ -2455,7 +2455,7 @@ class _AuthoringRenderer:
                 into.append(
                     self._indent(indent)
                     + f"{desired_name} = arith.constant {self._format_constant(expr.value, expr.type)} : "
-                    f"{self._render_type(expr.type)}"
+                    f"{self._render_arith_constant_type(expr.type)}"
                 )
                 return _RenderedValue(name=desired_name, type=expr.type)
             return _RenderedValue(
@@ -3525,7 +3525,7 @@ class _AuthoringRenderer:
             into.append(
                 self._indent(indent)
                 + f"{desired_name} = arith.constant {self._format_constant(value, expr.type)} : "
-                f"{self._render_type(expr.type)}"
+                f"{self._render_arith_constant_type(expr.type)}"
             )
             return _RenderedValue(name=desired_name, type=expr.type)
         return _RenderedValue(
@@ -3652,7 +3652,8 @@ class _AuthoringRenderer:
         self._constant_cache[cache_key] = name
         self._constant_lines.append(
             self._indent(4)
-            + f"{name} = arith.constant {self._format_constant(value, ty)} : {self._render_type(ty)}"
+            + f"{name} = arith.constant {self._format_constant(value, ty)} : "
+            f"{self._render_arith_constant_type(ty)}"
         )
         return name
 
@@ -3695,6 +3696,16 @@ class _AuthoringRenderer:
                 return "1" if value else "0"
             return str(value)
         raise NotImplementedError(f"unsupported constant type {ty!r}")
+
+    def _render_arith_constant_type(self, ty: SemanticType) -> str:
+        if isinstance(ty, SemanticScalarType) and is_integer_dtype(ty.dtype):
+            width = integer_bitwidth(ty.dtype)
+            if width is None:
+                raise NotImplementedError(
+                    f"unsupported integer dtype {ty.dtype.name!r} for arith.constant emission"
+                )
+            return f"i{width}"
+        return self._render_type(ty)
 
     def _format_float_constant(self, value: float, dtype_name: str) -> str:
         # Emit stable bit-pattern literals for values that are parse-sensitive

--- a/tilelang-dsl/tests/test_tilelang_dsl_v1.py
+++ b/tilelang-dsl/tests/test_tilelang_dsl_v1.py
@@ -2352,6 +2352,30 @@ class TileLangDSLDescriptorTests(unittest.TestCase):
 
         self.assertIn("PadValue.NULL.eval() is invalid", str(ctx.exception))
 
+    def test_unsigned_integer_constants_lower_with_signless_arith_types(self) -> None:
+        @pto.vkernel(op="tile_pad_value_ui32_max_eval_unique", dtypes=[(pto.ui32,)])
+        def kernel(tile: pto.Tile):
+            scalar = tile.pad_value.eval()
+            explicit = pto.ui32(4294967295)
+            return None
+
+        specialized = kernel.specialize(
+            tile=pto.TileSpecialization(
+                shape=(8, 16),
+                memory_space=pto.MemorySpace.UB,
+                config=pto.TileConfig.from_mapping(
+                    {
+                        "pad_value": pto.PadValue.MAX,
+                    }
+                ),
+            )
+        )
+
+        text = specialized.mlir_text()
+        self.assertIn("dtype=ui32", text)
+        self.assertIn("arith.constant 4294967295 : i32", text)
+        self.assertNotIn("arith.constant 4294967295 : ui32", text)
+
 
     def test_make_mask_vlds_vsts_and_vector_families_lower_inside_strict_vecscope(self) -> None:
         @pto.vkernel(op="eltwise", dtypes=[(pto.f32, pto.f32)], advanced=True)
@@ -2523,7 +2547,8 @@ class TileLangDSLDescriptorTests(unittest.TestCase):
         text = specialized.mlir_text()
         self.assertRegex(text, r"= arith\.floordivsi %in_gate_\d+, %c2_i32 : i32")
         self.assertRegex(text, r"= arith\.remsi %in_gate_\d+, %c7_i32 : i32")
-        self.assertRegex(text, r"= arith\.mulf %tmp_\d+, %c0\.5_f32 : f32")
+        self.assertRegex(text, r"%c0_5_f32 = arith\.constant 0\.5 : f32")
+        self.assertRegex(text, r"= arith\.mulf %tmp_\d+, %c0_5_f32 : f32")
         self.assertRegex(text, r"= arith\.addf %tmp_\d+, %tmp_\d+ : f32")
 
     def test_index_floordiv_lowers_to_divui_instead_of_floordivsi(self) -> None:

